packages/graphql-hooks/src/LocalGraphQLClient.js

```diff
@@ -0,0 +1,54 @@
+import GraphQLClient from './GraphQLClient'
+
+/** Local version of the GraphQLClient which only returns specified queries
+ * Meant to be used as a way to easily mock and test queries during development. This client never contacts any actual server.
+ * Queries are given in the form of an object of functions.
+ * Example:
+ ```
+    const localQueries = {
+      [allPostsQuery]: () => ({
+        allPosts: [
+          {
+            id: 1,
+            title: 'Test',
+            url: 'https://example.com'
+          }
+        ]
+      }),
+      [createPostMutation]: () => ({ createPost: { id: 1 } }),
+    }
+    const client = new LocalGraphQLClient({ localQueries })
+  ```
+ */
+class LocalGraphQLClient extends GraphQLClient {
+  constructor(config = {}) {
+    super(config)
+    this.localQueries = config.localQueries
```

Comment:
one reason why I'm not sure we need to extend the class is that I believe that the configuration options of the real client largely (or entirely even) don't apply to the local one. In fact I'd go as far as having the local client take a single constructor argument which is the local queries. am I missing something?

Reply:
Pretty much, but I think the difference is in how you might use it in testing/development code. Even if this does nothing, it will still have methods available to be called (if you look in the parent class, things like `setHeader`, or cache functions, or even the logging). This way we can have it behave as close as possible to the original one and it should be possible to run existing code by just replacing the "remote" client with a "local" one.

If we don't inherit, the local client might need to manually implement or mock more functions, which I think was what people were trying to avoid in the original PR

Reply:
Another option I thought about was to have a `RemoteGraphQLConnection` and `LocalGraphQLConnection`, and the client takes a `Connection` parameter, which would mean we're always using the real client and we're just replacing the object that takes care of sending requests.

I thought it was probably a bit overkill though so kept it simple for now

Reply:
In this case I have a feeling that what we're looking at is implementing an interface then, rather than inheriting a class, because the only thing that the local client needs from the real client is the interface and no implementation. 

Nevertheless, I suppose that doing it in that way is a bigger change, at least as far as types are concerned, so I'm good with inheritance as long as we agree that it's a compromise

Reply:
> I thought it was probably a bit overkill though so kept it simple for now

yes, not suggesting to do something overkill

Reply:
also this repo isn't in typescript, so there's no compiler to verify that the local client actually conforms to the interface of the real client easily

Reply:
but we have typings, don't we?

Reply:
no you're right!